compiler/resolution/lowerIterators.cpp

```diff
@@ -99,9 +99,15 @@ FnSymbol* getTheIteratorFnFromIteratorRec(Type* irType)
 FnSymbol* debugGetTheIteratorFn(Type* type) {
   FnSymbol* result = NULL;
   if (AggregateType* agg = toAggregateType(type)) {
-    if (agg->symbol->hasFlag(FLAG_TUPLE))
-      result = getTheIteratorFn(agg);
-    else if (agg->symbol->hasFlag(FLAG_ITERATOR_CLASS))
+    if (agg->symbol->hasFlag(FLAG_TUPLE)) {
+      // Sometimes the first field of the tuple is its size.
```

Comment:
I think that's always true during resolution, isn't it?

Reply:
Good point. I used to call this helper post-resolution, so this had not been an issue. It becomes an issue when I call whocalls() during resolution, which calls debugGetTheIteratorFn(), which crashes because of the `size` field.

With the above change it doesn't crash any longer.

I will update the comment to clarify: `Through the end of resolution, the first field is the tuple size.`